packages/backend/src/core/discovery/analyzeItem.ts

```diff
@@ -34,17 +41,13 @@ export async function analyzeItem(
     logger.proxyDetectionFailed(overrides.proxyType)
   }
 
-  const meta = await getMetadata(
-    provider,
+  const sourceCodeService = new SourceCodeService(provider)
```

Comment:
I understand that we will want to DI SourceCodeService later on, because rn it doesn't really give us many benefits.

Reply:
Yeah, this is the plan